ui/library/src/utils/scroll.ts

```diff
@@ -78,13 +71,60 @@ export function scrollToPosition(
     leftPixels = (width * (PDF_HEIGHT_POINTS - bottomPoints)) / PDF_HEIGHT_POINTS;
   }
 
-  targetDiv.scrollTo({
+  const pageId = generatePageIdFromIndex(pageIndex);
+
+  if (!pageId) {
+    return;
+  }
+
+  const pageIdElement = document.getElementById(pageId);
+
+  if (!pageIdElement) {
+    return;
+  }
+
+  const parentElement = pageIdElement.parentElement;
```

Comment:
was trying to create a function to grab all the parent elements but some how that crash the app so i just determined between `page-list` and `reader__main`

Reply:
Yeah, that's not going to do it. That assumes the parent of the page is scrollable...which it might not be. If you Google "determine first scrollable parent", the [first link gives you this algo](https://thewebdev.info/2021/06/27/how-to-find-the-first-scrollable-parent-element-with-javascript/#:~:text=To%20find%20the%20first%20scrollable%20parent%20element%20with%20JavaScript%2C%20we,has%20scrollHeight%20bigger%20than%20clientHeight%20.&text=to%20add%20the%20getScrollParent%20function,scrollHeight%20is%20bigger%20than%20node.):

```js
const getScrollParent = (node) => {
  if (node === null) {
    return null;
  }

  if (node.scrollHeight > node.clientHeight) {
    return node;
  } else {
    return getScrollParent(node.parentNode);
  }
}

const span = document.querySelector('span')
const scrollable = getScrollParent(span)
console.log(scrollable)
```

If it's crashing, then it's likely finding the wrong one. You might want an safety check, like:
```js
  if (node === document.documentElement) {
    return node;
  }
```

Alternatively, you could make sure it always returns something, like this:
```js
  if (!node) {
    return document.documentElement;
  }
```

So, a helper function like this might be the best option here:
```js
function getScrollParent(node) {
  if (typeof document === 'undefined') {
    return null;
  }
  if (!node) {
    return document.documentElement;
  }
  if (node.scrollHeight > node.clientHeight) {
    return node;
  }
  return getScrollParent(node.parentNode);
}
```